src/components/SecondaryPanes/XHRBreakpoints.js

```diff
@@ -103,19 +105,43 @@ class XHRBreakpoints extends Component<Props, State> {
     this.setState({ inputValue: target.value });
   };
 
-  hideInput = () => {
+  handleMethodChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    const target = e.target;
     this.setState({
-      focused: false,
-      editing: false,
-      editIndex: -1,
-      inputValue: "",
-      inputMethod: ""
+      focused: true,
+      editing: true,
+      inputMethod: target.value
     });
-    this.props.onXHRAdded();
+  };
+
+  hideInput = () => {
+    if (this.state.clickedOnOptions) {
+      this.setState({
+        focused: true,
+        clickedOnOptions: false
+      });
+    } else {
+      this.setState({
+        focused: false,
+        editing: false,
+        editIndex: -1,
+        inputValue: "",
+        inputMethod: "ANY"
+      });
+      this.props.onXHRAdded();
+    }
+  };
+
+  keepFocusOnInputButNoEdit = () => {
+    this.setState({ focused: true, editing: false });
   };
 
   onFocus = () => {
-    this.setState({ focused: true });
+    this.setState({ focused: true, editing: true });
+  };
+
+  onMouseDown = e => {
```

Comment:
@notjaril Just a nitpick, but I believe passing in this 'e' parameter is not necessary.

Reply:
good catch - thanks!